Adds backend widgets content fixing #1307

rber474 opened the pull request.

rber474 committed:
Adds backend widgets content fixing #1307

rber474 committed:
Adds backend widgets content fixing #1307

rber474 committed:
Merge branch '6.0' into widgets

rber474 commented:
Test failed because of the broken link https://52.demo.plone.org in install/index.md reported in issue #1553 and commented  by @stevepiercy days ago in https://community.plone.org/t/5-2-demo-is-down/18059?u=rber474

stevepiercy commented:
We can ignore the failed link until a decision is made about what to do with that site.

stevepiercy committed:
Clean up original version.
- Change from passive to active voice.
- Address the reader directly as "you".
- Clean up link references.
- One sentence per line.
- Clean up MyST syntax for inline literals and code blocks.
- Remove Unicode text markers for Python 3 code.
- Use double instead of single quotes for Python strings and other bits.

stevepiercy committed:
Clean up original version.
- Change from passive to active voice.
- Address the reader directly as "you".
- Clean up link references.
- One sentence per line.
- Clean up MyST syntax for inline literals and code blocks.
- Remove Unicode text markers for Python 3 code.
- Use double instead of single quotes for Python strings and other bits.

stevepiercy commented on docs/backend/widgets.md:
```diff
@@ -12,15 +12,652 @@ myst:
+
+            # widget.template is a template factory -
+            # Widget.render() will associate later this factory with the widget
+            widget.template = Z3ViewPageTemplateFile("templates/sections.pt")
```
Should this filename match the one give above, `yourwidget.pt`?

stevepiercy commented on docs/backend/widgets.md:
```diff
@@ -12,15 +12,652 @@ myst:
+You might want to customize this widget frame for your own form.
+Below is an example of how to do it.
+
+Copy [`widget.pt`](https://github.com/plone/plone.app.z3cform/blob/master/plone/app/z3cform/templates/widget.pt) to your own package, and edit it.
```
The subsequent code example names a template as `demo-widget.pt`. Should this line mention that name?

Also what is the file path to the template in the package?

stevepiercy commented on docs/backend/widgets.md:
````diff
@@ -12,15 +12,652 @@ myst:
+            zope.interface.alsoProvides(widget, IFilterWidget)
+```
+
+And then the page template which renders both 0. widget (min) and 1. widget (max) on the same line.
````
I have no idea what this is supposed to mean or say. Please clean this up to clarify the subsequent code example.

rber474 replied:
Would this be better?

`And then create the page template minmax.pt in the same directory of your form module and paste the following code which renders both widgets (min and max) in a single row.`

stevepiercy reviewed (changes requested):
This is excellent. I have a couple of comments and questions about the code examples. Please have a look. Thank you!

rber474 committed:
sync branch with 6.0

rber474 committed:
sync branch with 6.0

stevepiercy commented on docs/backend/widgets.md:
````diff
@@ -12,15 +12,652 @@ myst:
+            zope.interface.alsoProvides(widget, IFilterWidget)
+```
+
+And then the page template which renders both 0. widget (min) and 1. widget (max) on the same line.
````
Ah, now that makes sense. Let's use this.

```suggestion
Then create the page template `minmax.pt` in the same directory as your form module.
Paste the following code in this file.
The code renders both widgets, {guilabel}`min` and {guilabel}`max`, in a single row.
```

rber474 replied:
added suggestion

stevepiercy commented on docs/backend/widgets.md:
````diff
@@ -12,15 +12,652 @@ myst:
+
+And then the page template which renders both 0. widget (min) and 1. widget (max) on the same line.
+
+``` {code-block} html
````
The only reason to use `{codeblock}` is when you want to use Sphinx attributes, such as `emphasize-lines`.

````suggestion
```html
````

stevepiercy reviewed (changes requested):
Please remove the three submodules. We only update them when deploying the docs.

rber474 committed:
apply changes proposed by @stevepiercy

rber474 committed:
apply changes proposed by @stevepiercy

rber474 committed:
removes submodles

rber474 committed:
removes submodles

rber474 commented:
I am not sure if deleting the submodules won't be remove them in the main branch. Please, review that @stevepiercy

stevepiercy commented on docs/backend/widgets.md:
````diff
@@ -12,15 +12,688 @@ myst:
+``` {code-block} python
+:linenos:
+
+from z3c.form import form
````
Why are these added throughout the page? Line numbers don't add anything to these code examples. They just add clutter, and make `{code-block}` and an extra line break unnecessary. If you refer to a specific line number in the narrative text, or emphasize lines, then it makes sense to use the verbose form.

The above three lines (and elsewhere in this page) may be simplified as shown:

````
```python
````

rber474 replied:
didn't notice your commits. fixed again

stevepiercy commented on docs/backend/widgets.md:
```diff
@@ -12,15 +12,688 @@ myst:
+
+By default, `z3c.form` picks a widget based on the type of your field.
+You can change the widget using the `widget` directive if you want users to enter or view data in a different format.
+For example, you can change the widget for the `human` ``boolean`` field to use "yes" and "no" radio buttons instead of its default checkbox:
```
"boolean" is not an inline literal.

Side note, we use only single backticks for inline literals in MyST or Markdown.

```suggestion
For example, you can change the widget for the `human` boolean field to use "yes" and "no" radio buttons instead of its default checkbox:
```

stevepiercy commented on docs/backend/widgets.md:
```diff
@@ -12,15 +12,688 @@ myst:
+
+Copy [`widget.pt`](https://github.com/plone/plone.app.z3cform/blob/master/plone/app/z3cform/templates/widget.pt) to your own package, rename it as ``demo-widget.pt`` and edit it.
+
+Then add the following code to `configure.zcml`. Remember to fix the path of the template according to your own paths.
```
One sentence per line. 

```suggestion
Then add the following code to `configure.zcml`.
Remember to fix the path of the template according to your own paths.
```

stevepiercy commented:
> I am not sure if deleting the submodules won't be remove them in the main branch. Please, review that @stevepiercy

I meant to remove the commits that updated the submodule tips. Sorry for the confusion. I think because this PR comes from your fork, things are different from what I understand. Go ahead and restore the submodule tips to the latest from `documentation/6.0`. Thank you!

rber474 committed:
Update docs/backend/widgets.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

rber474 committed:
Update docs/backend/widgets.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

rber474 committed:
Update docs/backend/widgets.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

rber474 committed:
Update docs/backend/widgets.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

rber474 committed:
fix code-block syntax

rber474 committed:
fix code-block syntax

rber474 committed:
Merge branch 'widgets' of github.com:rber474/plone.documentation into widgets

rber474 committed:
Revert "removes submodles"

This reverts commit 74b127ba6d1fbda87f18b1c2c29c36c8f3dc70f0.

rber474 committed:
Revert "sync branch with 6.0"

This reverts commit 126664760eaf20cddaab89cc02595c6ac9d2b9eb.

rber474 committed:
Revert any changes of the submodules

rber474 commented:
@stevepiercy Thanks for your patient! I think everything is fixed now.

stevepiercy commented on docs/backend/widgets.md:
```diff
@@ -12,15 +12,677 @@ myst:
+
+## Changing a field's widget
+
+You can change a the widget that you use for a field in several ways.
```
```suggestion
You can change the widget that you use for a field in several ways.
```

stevepiercy commented on docs/backend/widgets.md:
````diff
@@ -12,15 +12,677 @@ myst:
+Now you can override the template factory in the `updateWidgets()` method of your form class.
+
+```python
+
````
```suggestion
```

stevepiercy commented on docs/backend/widgets.md:
````diff
@@ -12,15 +12,677 @@ myst:
+Then create a new marker interface in Python code.
+
+```python
+
````
```suggestion
```

stevepiercy commented on docs/backend/widgets.md:
````diff
@@ -12,15 +12,677 @@ myst:
+Then apply this marker interface to your widgets in `form.update()`.
+
+```python
+
````
```suggestion
```

stevepiercy commented on docs/backend/widgets.md:
````diff
@@ -12,15 +12,677 @@ myst:
+The following example shows how to create an input widget with minimum and maximum values.
+
+```python
+
````
```suggestion
```

stevepiercy commented on docs/backend/widgets.md:
````diff
@@ -12,15 +12,677 @@ myst:
+
+```
+
+Then you set the {guilabel}`my_combined_field` widget template in `updateWidgets()`:
````
I made a mistake. This is just an inline literal, not a GUI element.

```suggestion
Then you set the `my_combined_field` widget template in `updateWidgets()`:
```

stevepiercy reviewed (changes requested):
Just a little tidying up and it's good to go. Thank you!

rber474 committed:
Actualizar widgets.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

rber474 committed:
Actualizar widgets.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

rber474 committed:
Actualizar widgets.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

rber474 committed:
Actualizar widgets.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

rber474 committed:
Actualizar widgets.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

rber474 committed:
Actualizar widgets.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

stevepiercy committed:
Trim extra white space from code samples.

stevepiercy committed:
Update submodule tips to current ones.

stevepiercy commented:
OK, I cleaned up some extra white spaces and updated the submodule tips to the current ones, instead of deleting them, so that they matched those in `6.0` and dropped off this PR. Let's see what the CI says.

stevepiercy commented:
The two broken links are known. I notified evee of their bad secure certificate yesterday, and we are waiting for someone to either bring up or take down the 5.2 demo.

stevepiercy commented:
Thank you!
